spinalcordtoolbox/utils/shell.py

```diff
@@ -132,32 +138,40 @@ def _construct_fsleyes_syntax(viewer, files, colormaps, minmax, opacities):
     return cmd
 
 
-def _construct_itksnap_syntax(viewer, files, colormaps):
-    cmd = viewer
-    overlay_files = []
+def _construct_itksnap_syntax(viewer, files, im_types):
+    # Split the image files into two categories: grayscale images (used for `-g`/`-o`) and seg images (used for `-s`)
+    gray_images = []
+    seg_images = []
     for i in range(len(files)):
-        # -g is the "main image" option, and we assume that this is the first image
-        # TODO: This assumption is brittle, and could easily break if images are passed in the wrong order.
-        if i == 0:
-            cmd += ' -g ' + files[i]
+        if not im_types:
+            gray_images.append(files[i])
         else:
-            # - SCT uses colormaps to color overlaid segmentations for FSLeyes, because FSLeyes doesn't have
-            #   any "segmentation" options, and so files must be distinguished by choosing colors manually.
-            # - But, itk-snap requires that you explicitly specify which files are segmentation files (which
-            #   results in a red overlay for binary segmentations, and a rainbow overlay for multi-labeled
-            #   segmentations).
-            # - So, here we make the assumption that any files that would have been coloured (via fsleyes) should
-            #   instead be passed as segmentations to itk-snap.
-            # - TODO: This assumption is somewhat brittle, and could break if a colormap is used for a
-            #   non-segmentation segmentation file. But, the alternative would be a much bigger rewrite of the
-            #   display_viewer_syntax function.
-            if colormaps and colormaps[i] != "gray":
-                cmd += ' -s ' + files[i]
-            # All extra non-segmentation files have to be passed together to the '-o' (overlay) option
+            colormap = IMTYPES_COLORMAP[im_types[i]]['itksnap']
+            if colormap == 'gray':
+                gray_images.append(files[i])
+            elif colormap == 'seg':
+                seg_images.append(files[i])
             else:
-                overlay_files.append(files[i])
-    if overlay_files:
-        cmd += ' -o ' + " ".join(overlay_files)
+                raise ValueError(f"ITKSnap does not support colormap '{colormap}'")
+
+    # Construct ITKSnap command based on image types
+    # 1. '-g' is the "main image" and it's mandatory: i.e. You can't just display a single seg image ('-s') by itself,
+    #    or else itksnap will throw this error:
+    #        "Error: Option -s must be used together with option -g"
+    # NB: `-g` really should be a grayscale image, but if there are no gray images, we fall back to using a seg image.
+    cmd = f"{viewer} -g "
+    if len(gray_images) > 0:
+        cmd += f"{gray_images.pop(0)}"
+    else:
+        cmd += f"{seg_images.pop(0)}"
```

Comment:
Maybe this is too tricky, but you can shorten this a bit, using a [boolean operation](https://docs.python.org/3/reference/expressions.html#boolean-operations):
```suggestion
    main_image = (gray_images or seg_images).pop(0)
    cmd = f"{viewer} -g {main_image}"
```

Reply:
This is really cool! I didn't realize this syntax was possible, but this is much cleaner. :hearts: 

```python
>>> gray_images = []
>>> seg_images = ['t2_seg.nii.gz']
>>> print(seg_images)
['t2_seg.nii.gz']
>>> (gray_images or seg_images).pop(0)
't2_seg.nii.gz'
>>> print(seg_images)
[]
```